sound/soc/sof/core.c

```diff
@@ -143,6 +145,8 @@ static int sof_probe_continue(struct snd_sof_dev *sdev)
 		goto fw_load_err;
 	}
 
+	sdev->fw_state = SOF_FW_BOOT_IN_PROGRESS;
+
```

Comment:
shouldn't we set BOOT_FAILED on errors?

Reply:
@kv2019i @plbossart PREPARE->IN_PROGRESS would not change much in terms of the suspend actions and in a way a bit misleading as well. If we never switch from PREPARE->IN_PROGRESS , according to the diagram, it indicates that prepare failed.

Reply:
whereas if I do add PREPARE->BOOT_FAILED, it would mean prepare succeeded but boot failed